engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/BpmnEventTypeTest.java

```diff
@@ -0,0 +1,679 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.bpmn;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.tuple;
+
+import io.camunda.zeebe.engine.util.EngineRule;
+import io.camunda.zeebe.model.bpmn.Bpmn;
+import io.camunda.zeebe.model.bpmn.BpmnModelInstance;
+import io.camunda.zeebe.model.bpmn.builder.EndEventBuilder;
+import io.camunda.zeebe.model.bpmn.builder.ServiceTaskBuilder;
+import io.camunda.zeebe.model.bpmn.builder.StartEventBuilder;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.intent.ProcessInstanceIntent;
+import io.camunda.zeebe.protocol.record.value.BpmnElementType;
+import io.camunda.zeebe.protocol.record.value.BpmnEventType;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.util.function.Consumer;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class BpmnEventTypeTest {
```

Comment:
Hi, @remcowesterhoud, It's sounds difficult for me, the  `.containsSubsequence`  part of each test case is different, and I want to show the whole life cycle with `eventType`, `bpmnElementType`, `Intent` for each test case not just the `eventType` of one special BpmnElementType.

Reply:
That's fine, we can also keep it like this.

I am slightly confused about the test cases we have here. To me the goal of this test is to verify we are setting the correct event type on the correct records. The first 2 tests (`shouldTriggerErrorEventOnBoundaryEvent` and `shouldCatchErrorEventsByErrorCode`) seem to be more about testing the boundary event behavior. I understand that we are verifying the event type as part of the records, but why do we need both these tests? Wouldn't just the first one be sufficient?

I think we should also rename the test cases so it's clear what the test should be testing. So instead of `shouldTriggerErrorEventOnBoundaryEvent` it could be something along the lines of `shouldSetEventTypeForBoundaryEvents`.

Sorry for all these question, I'm just slightly confused about the tests 😄 Let me know what you think.